Sources/SmokeHTTPClient/HTTPClientChannelInboundHandler.swift

```diff
@@ -107,10 +108,19 @@ public final class HTTPClientChannelInboundHandler: ChannelInboundHandler {
             responseHead = request
             Log.verbose("Request head received.")
         // This is part of the response body
-        case .body(let byteBuffer):
-            // store this part of the body
-            bodyParts.append(byteBuffer)
-            Log.verbose("Request body of \(byteBuffer.readableBytes) bytes received.")
+        case .body(var byteBuffer):
+            let byteBufferSize = byteBuffer.readableBytes
+            let newData = byteBuffer.readData(length: byteBufferSize)
+            
+            if var previousPartialBody = partialBody,
```

Comment:
Wouldn't this be `newPartialBody`, since you're appending `newData` to it and updating `partialBody` to its now-appended value?

Reply:
Yes, that does make more sense. Updated.